app/assets/javascripts/discourse/app/controllers/user-activity-bookmarks.js

```diff
@@ -100,9 +100,22 @@ export default Controller.extend({
     this.model.more_bookmarks_url = response.more_bookmarks_url;
 
     if (response.bookmarks) {
-      this.content.pushObjects(
-        response.bookmarks.map((bookmark) => Bookmark.create(bookmark))
+      let bookmarkModels = response.bookmarks.map((bookmark) =>
+        Bookmark.create(bookmark)
       );
+
+      bookmarkModels.forEach((bookmark) => {
```

Comment:
This is done so the topic status related info is only passed to the topic-status component so stuff like `bookmark.pinned` does not interfere with it.

Reply:
cant we just do all of this in one loop ?

```
const bookmarkModels = response.bookmarks.map((bookmark) => {
  const bookmarkModel = Bookmark.create(bookmark);
  bookmarkModel.topicStatus = EmberObject.create({
    closed: bookmark.closed,
    archived: bookmark.archived,
    is_warning: bookmark.is_warning,
    pinned: false,
    unpinned: false,
    invisible: bookmark.invisible,
  });
  return bookmarkModel;
});
```